### DISCUSSION TITLE
Grafana Infinity Datasource API trouble for session tokens

### DISCUSSION THREAD

**Discussion by leonejames1** [grafana/grafana-infinity-datasource#567] [Specific APIs] [answered]
Need help from one of the gurus here :)

Trying to use grafana with infinity datasource to connect to an openapi system using ouath.  using session tokens..
Here's my problem.  The grant_type is password.  The datasource uses 'client credentials.  I heard that the new datasource handles refresh_tokens, but I cannot seem to get a datasource to work.

I've attached their requirements with examples. Any assistance would be greatly appreciated.
Right now I've resorted to used curl and another system.
![2](https://user-images.githubusercontent.com/8580770/228036701-ecaf6334-baac-48a0-9f68-283ebf5c05cf.JPG)
![1](https://user-images.githubusercontent.com/8580770/228036702-0251fb54-7164-4e7f-ba93-c2d54ab8c19f.JPG)

**Comment by yesoreyeram** [comment, marked as answer]
Hiya.. password grant_type is not supported yet.

**Reply by leonejames1** [reply]
Darn.... gotcha... Thank you, and have a great day.

**Reply by prawitdtgo** [reply]
I cheer up for this feature. Thanks.
